backend/private-graph/graph/schema.resolvers.go

```diff
@@ -4619,9 +4619,12 @@ func (r *queryResolver) ClientIntegration(ctx context.Context, projectID int) (*
 	if err != nil {
 		return integration, e.Wrap(err, "error querying session for project")
 	}
-	integration.Integrated = true
-	integration.ResourceSecureID = &firstSession.SecureID
-	integration.CreatedAt = &firstSession.CreatedAt
+
+	if firstSession.ID != 0 {
```

Comment:
i see, sorry for the hassle. so changing the `.First(` to a `.Find(` means that the `gorm.ErrRecordNotFound` is never returned, even if there are no results. that makes the zero-value check a safer one since you don't have to worry about the gorm internals.

Reply:
Thanks for confirming. I do see in the [docs](https://gorm.io/docs/query.html) this callout:

> If you want to avoid the ErrRecordNotFound error, you could use Find like db.Limit(1).Find(&user), the Find method accepts both struct and slice data

Was hoping there was a way to support `ErrRecordNotFound` using this method but I guess this is the expected behavior.

I've opened up #4942 for our Q2 planning to track any GORM guardrails that can be put in place.

@ccschmitz - per this thread, we can remove the `if e.Is(err, gorm.ErrRecordNotFound) {` checks for clarity.

Reply:
Whoops, I removed it in some of the checks but not all - my bad. Will clean this up. Thanks for catching.